pinot-controller/src/main/java/com/linkedin/pinot/controller/helix/core/realtime/PinotLLCRealtimeSegmentManager.java

```diff
@@ -729,4 +729,23 @@ private ZNRecord makeZnRecordForNewSegment(String realtimeTableName, int numRepl
     newSegMetadata.setStatus(CommonConstants.Segment.Realtime.Status.IN_PROGRESS);
     return newSegMetadata.toZNRecord();
   }
+
+  // An instance is reporting that it has stopped consuming a kafka topic due to some error.
+  // Mark the state of the segment to be OFFLINE in idealstate.
+  public void segmentStoppedConsuming(final LLCSegmentName segmentName, final String instance) {
+    String rawTableName = segmentName.getTableName();
+    String realtimeTableName = TableNameBuilder.REALTIME_TABLE_NAME_BUILDER.forTable(rawTableName);
+    final String segmentNameStr = segmentName.getSegmentName();
+    HelixHelper.updateIdealState(_helixManager, realtimeTableName, new Function<IdealState, IdealState>() {
+      @Override
+      public IdealState apply(IdealState idealState) {
+        idealState.setPartitionState(segmentNameStr, instance,
+            CommonConstants.Helix.StateModel.SegmentOnlineOfflineStateModel.OFFLINE);
+        Map<String, String> instanceStateMap = idealState.getInstanceStateMap(segmentNameStr);
+        LOGGER.info("Attempting to mark {} offline. Current map:{}", segmentNameStr, instanceStateMap.toString());
```

Comment:
I think it is useful to have and because it is in a path that is rarely executed, we will benefit to know that there were retries in updating helix. I will leave it in  there.